Neos.Media.Browser/Classes/Controller/Module/Management/AssetController.php

```diff
@@ -127,15 +133,28 @@ public function initializeObject()
      */
     public function deleteAction(\TYPO3\Media\Domain\Model\Asset $asset)
     {
-        try {
-            $this->assetService->getRepository($asset)->remove($asset);
-            $this->addFlashMessage('assetHasBeenDeleted', '', Message::SEVERITY_OK, [$asset->getLabel()], 1412375050);
-        } catch (AssetServiceException $exception) {
-            $message = $this->translator->translateById('media.deleteRelatedNodes', [], null, null, 'Modules', 'TYPO3.Neos');
+        $relationMap = [];
+        $relationMap[TypeHandling::getTypeForValue($asset)] = array($this->persistenceManager->getIdentifierByObject($asset));
+
+        if ($asset instanceof \TYPO3\Media\Domain\Model\Image) {
+            foreach ($asset->getVariants() as $variant) {
+                $type = TypeHandling::getTypeForValue($variant);
+                if (!isset($relationMap[$type])) {
+                    $relationMap[$type] = [];
+                }
+                $relationMap[$type][] = $this->persistenceManager->getIdentifierByObject($variant);
+            }
+        }
 
-            $this->addFlashMessage($message, '', Message::SEVERITY_WARNING, [], 1412422767);
+        $relatedNodes = $this->nodeDataRepository->findNodesByRelatedEntities($relationMap);
+        if (count($relatedNodes) > 0) {
+            $this->addFlashMessage('Asset could not be deleted, because there are still Nodes using it.', '', Message::SEVERITY_WARNING, array(), 1412422767);
+            $this->redirect('index');
         }
 
+        // FIXME: Resources are not deleted, because we cannot be sure that the resource isn't used anywhere else.
+        $this->assetRepository->remove($asset);
+        $this->addFlashMessage(sprintf('Asset "%s" has been deleted.', $asset->getLabel()), null, null, array(), 1412375050);
         $this->redirect('index');
     }
 
```

Comment:
we don't need this class do we? everything should fit in one controller now right?